### PR TITLE
Set up testing environment

### PR DESCRIPTION
**Pull request by Carltesio**
# [Set up testing environment for API](https://www.pivotaltracker.com/n/projects/2438205)
## Changes proposed in this pull request:
* Installing Rspec for unit and request spec
* Adding Shoulda matchers
* Configuring Factory Bot
* Adding Rails Pry

## Things I learned
* Refreshed knowledge about implementing and testing Devise Token Auth

## Packages/Gems added
- gem 'rack-cors'
- gem 'rspec-rails'
- gem 'shoulda-matchers'
- gem 'factory_bot_rails'

N.B.
Added also ci.rake file to be able to go on with the configuring of Coveralls and Semaphore.

### CODE REVIEW

File: .rspec [hunk 1]
```diff
@@ -0,0 +1,4 @@
+--require spec_helper
```
Comment (tochman) [inline .rspec]: please remove this. It's one or the other (`rails_helper` or `spec_helper`)

File: log/development.log [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  [1m[35m (0.6ms)[0m  [1m[34mSELECT pg_try_advisory_lock(8100797714064892305)[0m
+  [1m[35m (0.5ms)[0m  [1m[34mSELECT "schema_migrations"."version" FROM "schema_migrations" ORDER BY "schema_migrations"."version" ASC[0m
+  [1m[36mActiveRecord::InternalMetadata Load (0.5ms)[0m  [1m[34mSELECT "ar_internal_metadata".* FROM "ar_internal_metadata" WHERE "ar_internal_metadata"."key" = $1 LIMIT $2[0m  [["key", "environment"], ["LIMIT", 1]]
+  [1m[35m (0.2ms)[0m  [1m[35mBEGIN[0m
+  [1m[36mActiveRecord::InternalMetadata Create (0.7ms)[0m  [1m[32mINSERT INTO "ar_internal_metadata" ("key", "value", "created_at", "updated_at") VALUES ($1, $2, $3, $4) RETURNING "key"[0m  [["key", "environment"], ["value", "development"], ["created_at", "2020-03-10 19:42:56.231172"], ["updated_at", "2020-03-10 19:42:56.231172"]]
+  [1m[35m (2.5ms)[0m  [1m[35mCOMMIT[0m
+  [1m[35m (0.7ms)[0m  [1m[34mSELECT pg_advisory_unlock(8100797714064892305)[0m
+  [1m[35m (0.5ms)[0m  [1m[34mSELECT "schema_migrations"."version" FROM "schema_migrations" ORDER BY "schema_migrations"."version" ASC[0m
```
Comment (tochman) [inline log/development.log]: please remove this log from version control. What happened to your `.gitignore`? All log files are ignored by default when you create a Rails app.

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+ENV['RAILS_ENV'] ||= 'test'
```
Comment (tochman) [inline spec/rails_helper.rb:4]: Aren't you supposed to add coveralls settings here? I see that you included the gem, so I assume that you want to set it up as well....

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+Dir[Rails.root.join('spec/support/**/*.rb')].each { |f| require f }
+
+RSpec.configure do |config|
+  config.fixture_path = "#{::Rails.root}/spec/fixtures"
+  config.use_transactional_fixtures = true
+  config.infer_spec_type_from_file_location!
+  config.filter_rails_from_backtrace!
+  config.include FactoryBot::Syntax::Methods
+  config.include(Shoulda::Matchers::ActiveRecord, type: :model)
```
Comment (tochman) [inline spec/rails_helper.rb]: Lose the parenthesis. Be consistent. See line above.

**Review by tochman — CHANGES_REQUESTED** [review]
Make sure to remove `tmp/cache/bootsnap-load-path-cache `from version control. The `tmp` folder needs to be ignored.

File: .rspec (hunk 1 above)
Comment (janko101) [reply .rspec]: Understood. It's removed

File: log/development.log (hunk 2 above)
Comment (janko101) [reply log/development.log]: .gitignore wasn't included when we created a Rails app. We created it manually. I've updated the .gitignore file to look like the same as when it's automatically generated. Hope that fixes the issue.

File: spec/rails_helper.rb (hunk 3 above)
Comment (janko101) [reply spec/rails_helper.rb:4]: Yes we are and I was sure we did that, but apparently not. Apologies for that. It's added now

File: spec/rails_helper.rb (hunk 4 above)
Comment (janko101) [reply spec/rails_helper.rb]: Parentheses are forever lost.

**Pull request merged**